tmt/base.py

```diff
@@ -847,7 +847,7 @@ def _filters_conditions(self, nodes, filters, conditions):
             except fmf.utils.FilterError as error:
                 # Handle missing attributes as if condition failed
                 continue
-            except SyntaxError as error:
+            except Exception as error:
                 raise tmt.utils.GeneralError(
                     f"Invalid condition syntax: {error}")
```

Comment:
It's not only a syntax problem  any more. Error message will be misleading a bit for some cases. How about ~
```suggestion
                    f"Invalid --condition raised exception: {error}")
```